tests/console/yast2_samba.pm

```diff
@@ -14,258 +14,352 @@ use strict;
-    systemctl "show -p ActiveState slapd.service | grep ActiveState=active";
+    # change to not use SSL or TLS
+    wait_screen_change { send_key 'alt-u' };
+    send_key 'up';
+    assert_screen 'yast2_samba-server_ldap_advanced_expert_settings_not-use-ssl';
+    wait_screen_change { send_key 'ret' };
+    wait_screen_change { send_key 'alt-o' };
+
+    # now run test connection
+    send_key 'alt-t';
+    assert_screen 'yast2_samba-server_ldap_test-connection';
+    wait_screen_change { send_key 'alt-o' };
 }
 
-sub gui_current {
+sub setup_samba {
+    script_run("yast2 samba-server; echo yast2-samba-server-status-\$? > /dev/$serialdev", 0);
+
     # samba-server configuration for SLE older than 15 or opensuse TW
     assert_screen([qw(yast2_samba_installation yast2_still_susefirewall2)], 60);
     if (match_has_tag 'yast2_still_susefirewall2') {
         send_key 'alt-c';
     }
+
     wait_screen_change { send_key 'alt-w' };
     for (1 .. 12) { send_key 'backspace'; }
     # give a new name for Workgroup
-    type_string 'QA-Workgroup';
+    type_string($samba_directives->{workgroup});
     assert_screen 'yast2_samba-server_workgroup_new';
     send_key 'alt-n';
 
-    # select "Not a Domain Controller"
-    assert_screen 'yast2_samba_server_selection';
-    send_key 'alt-c';
+    ## Handle Domain Controller in sle12
+    if ((is_sle("<15")) || (is_leap("<15.0"))) {
+        # select "Not a Domain Controller"
+        assert_screen 'yast2_samba_server_selection';
+        send_key 'alt-c';
 
-    # check "Not a DC" is select
-    assert_screen 'yast2_samba-server_not-a-dc_selected';
-    send_key 'alt-n';
-    wait_still_screen;
-    # now move to Samba configuration and check enable service start during boot
-    send_key 'alt-r';
-    assert_screen 'yast2_samber-server_start-during-boot';
+        # check "Not a DC" is select
+        assert_screen 'yast2_samba-server_not-a-dc_selected';
+        send_key 'alt-n';
+    }
 
-    # open port in firewall if it is enabled and check network interfaces, check long text by send key right.
-    assert_screen 'yast2_samba_open_port_firewall';
+    # service starts during boot, wait to load default data
+    assert_screen 'yast2_samba-startup-configuration';
+    wait_screen_change { send_key 'alt-r' };
+    assert_screen 'yast2_samba-server_start-during-boot';
+    # open firewalld port
     wait_screen_change { send_key 'alt-f' };
-    send_key 'alt-i';
-
-    assert_screen 'yast2_samba_firewall_port_details';
-    wait_screen_change { send_key 'alt-e' };
-    wait_screen_change { send_key 'alt-a' };
-    wait_screen_change { send_key 'alt-o' };
+    assert_screen 'yast2_samba_open_port_firewall';
 
-    # switch to Samba Configuration - Shares
-    send_key 'alt-s';
+    ## switch to Samba Configuration - Shares
+    wait_screen_change { send_key 'alt-s' };
+    wait_still_screen(2);
     assert_screen 'yast2_samba-server_shares';
```

Comment:
I don't understand. Can you explain why three screen checks need to be done? I mean first `wait_screen_change`, then wait for still screen and then the final `assert_screen`. What could happen if we just keep the `assert_screen`?

Reply:
``wait_screen_change`` can be dismissed but the ``wait_still_screen(2);`` is there to ensure that yast will load default data properly. Otherwise the script might miss some configuration. I have faced that same issue with the ftp test. Test was trying to modify fields which where "grayed" because yast was still loading default data for the screen.